src/systemTest/java/com/flightstats/hub/system/functional/ChannelMaxItemsEnforcementTest.java

```diff
@@ -0,0 +1,106 @@
+package com.flightstats.hub.system.functional;
+
+import com.flightstats.hub.model.ChannelConfig;
+import com.flightstats.hub.model.ChannelType;
+import com.flightstats.hub.system.extension.TestClassWrapper;
+import com.flightstats.hub.system.service.ChannelConfigService;
+import com.flightstats.hub.system.service.ChannelItemCreator;
+import com.flightstats.hub.system.service.ChannelItemRetriever;
+import com.flightstats.hub.system.service.HubInternalService;
+import com.flightstats.hub.system.service.S3Service;
+import lombok.SneakyThrows;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+
+@Execution(ExecutionMode.SAME_THREAD)
+class ChannelMaxItemsEnforcementTest extends TestClassWrapper {
+    private static final String TEST_DATA = "TEST_DATA";
+    private final List<String> items = new ArrayList<>();
+    private String channelName;
+    @Inject
+    private ChannelConfigService channelConfigService;
+    @Inject
+    private ChannelItemCreator channelItemCreator;
+    @Inject
+    private ChannelItemRetriever channelItemRetriever;
+    @Inject
+    private HubInternalService hubInternalService;
+    @Inject
+    private S3Service s3Service;
+
+    @AfterEach
+    void cleanup() {
+        items.clear();
+        channelConfigService.delete(channelName);
+    }
+
+    @SneakyThrows
+    private void addItemsOverTime() {
+        /*
+            batch items are compared with granularity of one minute
+            so if the excess items are of the same minute they will not be deleted
+        */
+        items.add(channelItemCreator.addItem(channelName, TEST_DATA));
+        Thread.sleep(TimeUnit.MINUTES.toMillis(1));
```

Comment:
Bummer to wait a minute on a test doing nothing :(. Not sure what a good solution is though. Maybe an optional channelItemCreator constructor arg or backdoor setter to hack how it decides the time of the item added, so you do something like:
```
items.add(channelItemCreator.addItem(channelName, TEST_DATA))
channelItemCreator.hackAddThisTimeToAddItem(TimeUnits.MINUTES.toMillis(1));
items.addAll(....);
```

Reply:
I was going to suggest using a historical channel, but I think that prevents us from changing over to have a max items setting...which means we'd have to mess with dynamo directly to update the config.

Reply:
Without knowing much about the hub's codebase, is there a way to use a Clock object? ie, in production you use the system clock, but in test you use a mutable clock?

Reply:
+1 to Lucas' suggestion.
We started from the Guava Ticker because it fit some other use cases but a convertible time source that is used globally across the application has a lot of benefits.  Happy to share pointers to our various implementations if desired but if I was doing it today from scratch I would probably expose getInstant() and push all use towards that contract myself.

Reply:
By Clock object I mean a `java.time.Clock`

Reply:
I think moving that way would be great, but I'm hesitant to mess with the way hub deals with time before we have holistic tests that make sure we don't break the current implementation or emergent(/undocumented) behavior.

Reply:
> I think moving that way would be great, but I'm hesitant to mess with the way hub deals with time before we have holistic tests that make sure we don't break the current implementation or emergent(/undocumented) behavior.

I agree with @lkemmerer about getting into tinkering with the hub's time implementation, though long term those ideas would make testing things like this more ergonomic and stable. We have had a conversation offline about this as well. In this particular case I think doing too much to work around this would amount to distracting from what is a (albeit minor) bug.